### ISSUE TITLE
circleci/aws-s3@1.0.10 fails with overwrite: true

### ISSUE THREAD

**Issue by jams2** [CircleCI-Public/circleci-orbs#168]
### Orb version

1.0.10

### What happened

Experiencing here a similar issue to #156 , but the syntax of our config matches the documentation. Running `aws-s3/sync` with `overwrite: true` causes the following error:

```
/bin/bash: line 5: --delete: command not found
Exited with code 127
```

The failed job is here: https://circleci.com/gh/thebloggerprogramme/frontend/23

Our top-level config looks like this:

```
version: 2.1
orbs:
  aws-s3: circleci/aws-s3@1.0.10
  heroku: circleci/heroku@0.0.8
```

Here's the <span id="job-config">config</span> for that job:

```
  deploy-staging:
    docker:
      - image: circleci/python:2.7
    working_directory: ~/build
    steps:
      - checkout
      - attach_workspace:
          at: ~/build
      - aws-s3/sync:
          from: dist/assets
          to: 's3://mybucket/test'
          arguments: |
            --acl public-read \
            --cache-control "max-age=86400"
          overwrite: true
``` 

Looking at the output in the job, it does look like a problem with the way the command is being constructed;

```
#!/bin/bash -eo pipefail
aws s3 sync \
  dist/assets s3://thebloggerprogramme-staging/test \
  --acl public-read \
--cache-control "max-age=86400"
 \
  --delete
```

CircleCI support initially reported that our config had two trailing backslashes after the `--acl public-read` option, like this;

```
   deploy-staging:
     docker:
       - image: circleci/python:2.7
     working_directory: ~/build
     steps:
       - checkout
       - attach_workspace:
           at: ~/build
       - aws-s3/sync:
           from: dist/assets
           to: 's3://mybucket/test'
           arguments: |
             --acl public-read \\
             --cache-control \"max-age=86400\"
           overwrite: true
```

but I can confirm that the config.yml that was checked in had a single trailing slash, as you can see in the job config [above](#job-config).

The sync works fine without the `overwrite` parameter. After spending a little time looking into this it turns out we don't need the `--delete` option but I thought it would be useful to raise the issue anyway.

**Comment by iynere** [comment]
looks like there's some kind of yaml parsing issue where a newline is being inserted where it should be.... i wonder if we can just make `arguments` the last portion of the command & that would take care of it...

**Comment by jams2** [comment]
You're right [(source)](https://github.com/CircleCI-Public/circleci-orbs/blob/98bc65476d06c3d00ed561813291019fa03426b1/src/aws-s3/orb.yml#L107)
```yaml
aws s3 sync \
      <<parameters.from>> <<parameters.to>> \
      <<#parameters.arguments>><<parameters.arguments>><</parameters.arguments>><<#parameters.overwrite>> \
      --delete<</parameters.overwrite>>
```

A newline is being inserted after `<<parameters.arguments>>`, and it looks like the cause of that, taking my config above:

```yaml
arguments: |
    --acl public-read \
    --cache-control "max-age=86400"
```

is the expansion including the unescaped newline after `--cache-control`. Moving `arguments` to the end is probably the simplest solution. I don't mind testing and throwing a PR together some time this week if you like.

**Comment by iynere** [comment]
it's such a small change, i can take care of it right now @jams2 

i'll merge this change to staging, which will publish a dev version w/ the change, & then, since we're not currently doing any extensive integration or usage testing for orbs in this repo (hard to do in the monorepo—our single-repo orbs are tested much more extensively), can you give your job a shot again with the new dev version ?

**Comment by iynere** [comment]
ok, please give this a shot with `circleci/aws-s3@dev:staging-71c6a65` @jams2 

if that fixes your issue, i'll merge into master, which will trigger the release of a new semantic / production orb version

**Comment by jams2** [comment]
That fixes it.

Using config:

```
version: 2.1
orbs:
  aws-s3: circleci/aws-s3@dev:staging-71c6a65
jobs:
  deploy-staging:
    docker:
      - image: circleci/python:2.7
    working_directory: ~/repo
    steps:
      - run: mkdir syncdir && echo "test" >> syncdir/test.txt
      - aws-s3/sync:
          from: syncdir
          to: 's3://thebloggerprogramme-staging/test2'
          arguments: |
            --acl public-read \
            --cache-control "max-age=86400"
          overwrite: true

workflows:
  version: 2.1
  test_build:
    jobs:
      - deploy-staging
```

command is expanded as:

```
- deploy:
        name: Deploy to S3
        command: |+
          aws s3 sync \
            syncdir s3://thebloggerprogramme-staging/test2 --delete \
            --acl public-read \
          --cache-control "max-age=86400"
```

Job is [here](https://circleci.com/gh/jams2/orb-testing/4)

Thanks!

**Comment by iynere** [comment]
awesome ! will merge this to master 🙂 thank you !! @jams2

**Comment by iynere** [comment]
fixed in `circleci/aws-s3@1.0.11` 🎉
